chain/ton/provider/rpc_provider.go

```diff
@@ -122,6 +187,38 @@ func (p *RPCChainProvider) Initialize(ctx context.Context) (chain.BlockChain, er
 	return *p.chain, nil
 }
 
+// createLiteclientConnectionPool creates connection pool returning concrete type for production use
+func createLiteclientConnectionPool(ctx context.Context, liteserverURL string) (*liteclient.ConnectionPool, error) {
+	// Validate URL format first
+	if err := validateLiteserverURL(liteserverURL); err != nil {
+		return nil, err
+	}
+
+	// Parse the URL
+	urlPart := strings.TrimPrefix(liteserverURL, "liteserver://")
+	parts := strings.Split(urlPart, "@")
+	publicKey := parts[0]
+	hostPort := parts[1]
+
+	pool := liteclient.NewConnectionPool()
+
+	// Setup connection timeout - mirrors tonutils-go AddConnectionsFromConfig logic
+	timeout := 3 * time.Second
+	if dl, ok := ctx.Deadline(); ok {
+		timeout = time.Until(dl)
+	}
+
+	connCtx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	err := pool.AddConnection(connCtx, hostPort, publicKey)
```

Comment:
why not just this?
```suggestion
	err := pool.AddConnection(ctx, hostPort, publicKey)
```

if the `ctx` has its deadline, then it will be respected when it is pased into `pool.AddConnection`  instead of manually calculating it ?